spec/array_based_collection_spec.rb

```diff
@@ -0,0 +1,69 @@
+RSpec.describe QueryRelation::ArrayBasedCollection do
+  let(:model) do
+    Class.new do
+      extend QueryRelation::Queryable
+      extend QueryRelation::ArrayBasedCollection
+
+      def self.array
+        [
+          {:name => 'Zach', :age => 35},
+          {:name => 'Lisa', :age => 25},
+          {:name => 'Jon',  :age => 45},
+          {:name => 'Jon',  :age => 75},
+        ]
+      end
+
+      def self.search(mode, options = {})
+        records = filter_array_collection(array, options) #unless options.empty?
+        case mode
+        when :all   then records
+        when :first then records.first
+        when :last  then records.last
+        end
+      end
+
+      private_class_method :search
+    end
+  end
+
+  it ".all" do
+    relation = model.all
+    expect(relation).to be_kind_of(QueryRelation)
+    expect(relation.to_a).to eq(model.array)
+  end
+
+  it ".where with no results" do
+    relation = model.where(:a => 1)
+    expect(relation.to_a).to be_empty
+  end
+
+  it ".where with one result" do
+    relation = model.where(:name => 'Zach')
+    expect(relation.to_a).to eql([{:name => 'Zach', :age => 35}])
+  end
+
+  it ".where with multiple results" do
+    relation = model.where(:name => 'Jon')
```

Comment:
add a test for `model.where(:name => %w(John Zach)`

Reply:
Doesn't work atm.